Makefile

```diff
@@ -0,0 +1,81 @@
+LANG      := en
+LANG_FILE := $(shell test -f LANGS.md.bak && echo LANGS.md.bak || echo LANGS.md)
+LANG_DATA := $(shell grep "$(LANG)/" $(LANG_FILE))
+LANG_NAME := $(shell echo "$(LANG_DATA)" | sed 's/.*\[\(.*\)\].*/\1/')
+
+define ebook_support
+	@if which ebook-convert 1> /dev/null; then\
+		npx honkit $(1) ./ ./djangogirls.$(1);\
+	else\
+		echo "Error: ebook-convert is not found. Follow the guide at https://honkit.netlify.app/ebook";\
+	fi
+endef
+
+help:
+	@echo
+	@echo "Usage: make command [LANG=<language_short_code>]"
+	@echo
+	@echo "    LANG : Language short codes are found in LANGS.md. Default is 'en'(English)."
+	@echo "           ( Refer $(LANG_FILE) for short codes of Languages avaiable. )"
+	@echo
+	@echo "Commands:"
+	@echo "  help      - Display make command list."
+	@echo "  dev       - Setup project and start the development server with debugging enabled."
+	@echo "  check     - Check for root directory for various dependencies."
+	@echo "  setup     - Setup the temporary language and install node dependencies for the development. Default is 'en'"
+	@echo "  build     - Build the honkit project."
+	@echo "  build-dev - Build the honkit project with debug log."
+	@echo "  serve     - Start honkit server locally for development."
+	@echo "  pdf       - Generate PDF version of DjangoGirls tutorial."
+	@echo "  epub      - Generate EPUB version of DjangoGirls tutorial."
+	@echo "  mobi      - Generate MOBI version of DjangoGirls tutorial."
+	@echo "  mode      - Shows the development mode status."
+	@echo "  exit      - Reset the project and exit development mode."
+
+check: package.json book.json LANGS.md
+	@if ! which node 1> /dev/null; then\
+		echo "Node.js not found. Please install/reinstall Node.js. NVM is recommended for installation(https://github.com/nvm-sh/nvm).";\
```

Comment:
Can this somehow trigger a non-zero exit code so `make` will fail?

Reply:
we can use `$(error ERROR_MESSAGE)` instead of `echo`. That exit with non-zero.

Reply:
> we can use `$(error ERROR_MESSAGE)` instead of `echo`. That exit with non-zero.

Cool, I didn't know `make` [provided such functions](https://www.gnu.org/software/make/manual/html_node/Make-Control-Functions.html). Let's use that!

(According to [this StackOverflow answer](https://stackoverflow.com/a/29424978/674064) they are specific to _GNU make_, but I guess that's sufficiently acceptable for us. This Makefile is unlikely to be used with anything else.)

Reply:
I thought `$(error)` would work. But it did not work as required. It ignores any bash command nearby. So I used the bash command `false` to get the same result. Check [c2c3522](https://github.com/DjangoGirls/tutorial/pull/1729/commits/c2c352223c77ee6f1c4717dc59e33c3d459340f2).